File: java/bypstest-api/src/com/wilutions/byps/test/api/srvr/ServerIF.java
package com.wilutions.byps.test.api.srvr;

import java.io.InputStream;
import java.util.List;
import java.util.Set;

import com.wilutions.byps.BRemote;
import com.wilutions.byps.RemoteException;

public interface ServerIF extends BRemote {

	public int callClientIncrementInt(int v) throws RemoteException;
	
	public void setPartner(ClientIF client) throws RemoteException;
	
	public ClientIF getPartner() throws RemoteException;
	
	public List<InputStream> getStreamsFromClient() throws RemoteException;
	
	public void putStreamsOnClient(List<InputStream> streams) throws RemoteException;
	
	public void registerWithClientMap(int id) throws RemoteException;
	
	public ClientIF getClient(int id) throws RemoteException;
	
	public Set<Integer> getClientIds() throws RemoteException;

	public int callClientParallel(int v) throws RemoteException;
	
}

File: csharp/bypstest-ser/src-ser/byps/test/api/srvr/ServerIF.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace byps.test.api.srvr
{
	
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface ServerIF : BRemote {
	
		int CallClientIncrementInt(int v);
		void CallClientIncrementInt(int v, BAsyncResult<int> asyncResult) ;
		Task<int> CallClientIncrementIntAsync(int v);
		
		void SetPartner(ClientIF client);
		void SetPartner(ClientIF client, BAsyncResult<Object> asyncResult) ;
		Task SetPartnerAsync(ClientIF client);
		
		ClientIF GetPartner();
		void GetPartner(BAsyncResult<ClientIF> asyncResult) ;
		Task<ClientIF> GetPartnerAsync();
		
		IList<System.IO.Stream> GetStreamsFromClient();
		void GetStreamsFromClient(BAsyncResult<IList<System.IO.Stream>> asyncResult) ;
		Task<IList<System.IO.Stream>> GetStreamsFromClientAsync();
		
		void PutStreamsOnClient(IList<System.IO.Stream> streams);
		void PutStreamsOnClient(IList<System.IO.Stream> streams, BAsyncResult<Object> asyncResult) ;
		Task PutStreamsOnClientAsync(IList<System.IO.Stream> streams);
		
		void RegisterWithClientMap(int id);
		void RegisterWithClientMap(int id, BAsyncResult<Object> asyncResult) ;
		Task RegisterWithClientMapAsync(int id);
		
		ClientIF GetClient(int id);
		void GetClient(int id, BAsyncResult<ClientIF> asyncResult) ;
		Task<ClientIF> GetClientAsync(int id);
		
		ISet<int> GetClientIds();
		void GetClientIds(BAsyncResult<ISet<int>> asyncResult) ;
		Task<ISet<int>> GetClientIdsAsync();
		
		int CallClientParallel(int v);
		void CallClientParallel(int v, BAsyncResult<int> asyncResult) ;
		Task<int> CallClientParallelAsync(int v);
		
		
	}
}  // end namespace
